### PR TITLE
feat: refactor controller service

### PR DESCRIPTION
**Pull request by cr7258**
fixes: #8 

**1. Use [GlobalExceptionHandler](https://github.com/cr7258/quicksense-backend/blob/refactor-controller-service/src/main/java/pro/quicksense/exception/GlobalExceptionHandler.java) to handle errors uniformly.**
**2. Unified response body ([ApiResponse](https://github.com/cr7258/quicksense-backend/blob/refactor-controller-service/src/main/java/pro/quicksense/common/ApiResponse.java)), for example:**

success:
```
# request
curl "http://localhost:8080/user/register" -X POST -H "Content-Type: application/json" -d '
{
        "username": "seven",
        "password": "test123",
        "email": "chengzw258@163.com"
}'

# response
{
  "status": "CREATED",
  "timestamp": "2024-07-27 21:31:42",
  "message": "User registered successfully",
  "data": {
    "id": 1,
    "username": "seven",
    "password": "$2a$10$WSPzYFw.yRWqoVsk.HFdNuVPUv.C7If2yXUzrZqCUIQs6MSv7vPTG",
    "email": "chengzw258@163.com",
    "status": 1
  }
}
```
fail:
```bash
# request
curl "http://localhost:8080/user/register" -X POST -H "Content-Type: application/json" -d '
{
        "username": "seven",
        "password": "test123",
        "email": "chengzw258163.com"
}' 

# response
{
  "status": "UNPROCESSABLE_ENTITY",
  "timestamp": "2024-07-27 21:33:43",
  "message": "Method or Argument not valid",
  "data": [
    {
      "rejectedValue": "chengzw258163.com",
      "field": "email",
      "errorMessage": "Invalid email format",
      "object": "user"
    }
  ]
}
```
**3. Split login logic from UserService to [LoginService](https://github.com/cr7258/quicksense-backend/blob/refactor-controller-service/src/main/java/pro/quicksense/service/impl/LoginServiceImpl.java)**
**4.Introduce Spring Data Jpa to simplify database CRUD operations. (further reading: [findBy Methods in Spring Data JPA Repositories](https://www.geeksforgeeks.org/findby-methods-in-spring-data-jpa-repositories/))**

### CODE REVIEW

**Comment by cr7258** [comment]
Unit tests will be handled in this issue: https://github.com/cr7258/quicksense-backend/issues/7

File: src/main/java/pro/quicksense/QuicksenseApplication.java [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+@Slf4j
 public class QuicksenseApplication {
 
 	public static void main(String[] args) {
 		SpringApplication.run(QuicksenseApplication.class, args);
-		System.out.println("Hello World");
+		log.info("User Management System started successfully...");
```
Comment (Mchen1995) [inline src/main/java/pro/quicksense/QuicksenseApplication.java]: Springboot will print the message like: "Started QuicksenseApplication in 2.358 seconds (process running for 3.465)", this line of code could be removed.

File: src/main/java/pro/quicksense/service/impl/LoginServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Autowired
+    private CodeUtil codeUtil;
+
+    @Autowired
+    private JwtInterceptor jwtInterceptor;
+
+    @Override
+    public String loginByEmail(EmailLogin emailLogin) {
+        if (!codeUtil.verifyCode(emailLogin.getEmail(), emailLogin.getVerifyCode(), CommonConstant.KEY_PREFIX)) {
```
Comment (Mchen1995) [inline src/main/java/pro/quicksense/service/impl/LoginServiceImpl.java:29]: What if the email address entered by the user has not been registered yet?

File: src/main/java/pro/quicksense/controller/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@RestController
+@RequestMapping("/login")
+public class LoginController {
+
+    @Autowired
+    private LoginService loginService;
+
+    @PostMapping("/username")
+    public ResponseEntity<Object> loginByUsername(@RequestBody @Valid UsernameLogin usernameLogin) {
+        return ApiResponseBuilder.success(HttpStatus.OK, "Username login successful", loginService.loginByUsername(usernameLogin));
```
Comment (Mchen1995) [inline src/main/java/pro/quicksense/controller/LoginController.java:25]: Always returns a 'success' message?

File: src/main/java/pro/quicksense/QuicksenseApplication.java (hunk 1 above)
Comment (cr7258) [reply src/main/java/pro/quicksense/QuicksenseApplication.java]: removed

**Pull request merged**